File: 75. Sort Colors/Python/Sort Colors.py
class Solution(object):
    def sortColors(self, nums):
        """
        :type nums: List[int]
        :rtype: void Do not return anything, modify nums in-place instead.
        """
        left = 0
        while left < len(nums) and nums[left] == 0:
            left += 1
        right = len(nums) - 1
        while right >= 0 and nums[right] == 2:
            right -= 1
        current = left
        while current <= right and left <= right:
            if nums[current] == 0:
                if current == left:
                    current += 1
                else:
                    nums[left], nums[current] = nums[current], nums[left]
                left += 1
            elif nums[current] == 2:
                if current == right:
                    current += 1
                else:
                    nums[right], nums[current] = nums[current], nums[right]
                right -= 1
            else:
                current += 1

File: 75. Sort Colors/Java/Sort Colors.java
public class Solution {
    public void sortColors(int[] nums) {
        int left = 0;
        while (left < nums.length && nums[left] == 0) {
            left++;
        }
        int right = nums.length - 1;
        while (right >= 0 && nums[right] == 2) {
            right--;
        }
        int current = left;
        while (current <= right && left <= right) {
            if (nums[current] == 0) {
                if (current == left) {
                    current++;
                } else {
                    int temp = nums[left];
                    nums[left] = nums[current];
                    nums[current] = temp;
                }
                left++;
            } else if (nums[current] == 2) {
                if (current == right) {
                    current++;
                } else {
                    int temp = nums[right];
                    nums[right] = nums[current];
                    nums[current] = temp;
                }
                right--;
            } else {
                current++;
            }
        }
    }
}